backend/core/src/core/tag/application/tag-service.ts

```diff
@@ -0,0 +1,65 @@
+import { inject, injectable } from 'inversify';
+import { CONTAINER_TYPES } from '~/shared/types/types';
+
+import { SearchByTagResponseDto, TagCreateRequestDto, TagResponseDto } from 'shared/build';
+
+import { TagRepository } from '~/core/tag/port/tag-repository';
+import { castToTagResponseDto } from './dtos/cast-to-tag-response-dto';
+import { VideoRepository } from '~/core/video/port/video-repository';
+import { castToSearchByTagResponseDto } from './dtos/cast-to-search-response-dto';
+
+@injectable()
+export class TagService {
+  private tagRepository: TagRepository;
+  private videoRepository: VideoRepository;
+
+  constructor(
+    @inject(CONTAINER_TYPES.TagRepository) tagRepository: TagRepository,
+    @inject(CONTAINER_TYPES.VideoRepository) videoRepository: VideoRepository,
+  ) {
+    this.tagRepository = tagRepository;
+    this.videoRepository = videoRepository;
+  }
+
+  async getByName({ name }: TagCreateRequestDto): Promise<TagResponseDto | null> {
+    const tag = await this.tagRepository.getByName(name);
+    return tag && castToTagResponseDto(tag);
+  }
+
+  async getById(id: string): Promise<TagResponseDto | null> {
+    const tag = await this.tagRepository.getById(id);
+    return tag && castToTagResponseDto(tag);
+  }
+
+  async search({
+    take,
+    skip,
+    tags,
+  }: {
+    take: number | undefined;
+    skip: number | undefined;
+    tags: string[];
+  }): Promise<SearchByTagResponseDto[]> {
+    const videos = await this.videoRepository.searchByTags({ take, skip, tags });
+
+    return videos.map((video) => castToSearchByTagResponseDto(video));
+  }
+
+  async createTag({ name }: TagCreateRequestDto): Promise<TagResponseDto | undefined> {
+    const isTagCreated = await this.tagRepository.getByName(name);
+    if (isTagCreated) {
+      return;
+    }
+    return this.tagRepository.createTag({ name });
+  }
```

Comment:
can you please rewrite it to createTags taking an array of tag names? otherwise user will have to send a separate request for every of the many tags they attach to the stream

Reply:
i`m not sure that it is possible to do in one sql query